internal/martian/flush.go

```diff
@@ -53,3 +54,31 @@ func (w flushAfterChunkWriter) WriteString(s string) (n int, err error) {
 	}
 	return
 }
+
+// flushAfterSSEWriter forces a flush after each '\n\n' is written.
+type flushAfterSSEWriter struct {
```

Comment:
We need a flushWriter that can be used in proxy conn and handler for sse and chunked encoding.

```
type interface flusher Flush() error

type afterFlusher struct {
  w io.Writer
  f flusher
  eom [2]byte
}
```

This needs to come with a set of extensive tests.

This needs to be applied everywhere.